pkg/windows/windows.go

```diff
@@ -857,15 +865,21 @@ func (vm *windows) ensureHNSNetworksAreRemoved() error {
 	// VIP HNS endpoint created by the operator is also deleted when the HNS networks are deleted.
 	for _, network := range []string{BaseOVNKubeOverlayNetwork, OVNKubeOverlayNetwork} {
 		err = wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
+			// reinitialize and retry on failure
 			if err := vm.removeHNSNetwork(network); err != nil {
-				return false, errors.Wrapf(err, "error removing %s HNS network", network)
+				vm.log.V(1).Error(err, "error removing %s HNS network", "network", network)
+				if err := vm.Reinitialize(); err != nil {
+					return false, errors.Wrapf(err, "error reinitializing VM after removing %s HNS network", network)
+				}
+				return false, nil
 			}
 			if err := vm.Reinitialize(); err != nil {
 				return false, errors.Wrapf(err, "error reinitializing VM after removing %s HNS network", network)
```

Comment:
by returning errors here on the two reinitialize calls the choice is being made not to retry. Is that on purpose? I would think it would be better to return false, nil and retry.